docs/modules/concepts/pages/configuration.adoc

```diff
@@ -0,0 +1,34 @@
+= Configuration
+
+== Overview
+
+An important aspect to understand about James is the idea of "configuration".
+Sometimes "configuration" means different things to different people, so we
+wanted to ensure that what we mean by the concept is clear and precise, 
+because in order operate a James server, you will necessarily need to 
+configure it.
+
+A "*Configuration*" is a https://martinfowler.com/bliki/ValueObject.html[value object]
```

Comment:
It's not factually wrong that it's a Value Object but I think it's not a good description of it. Value Object usually represents a single value (in Martin Fowler article, a Coordinate is a single value for example).
Configuration is IMO an immutable data structure.

Reply:
I know you know this, but just to establish a baseline for understanding...

Value objects and immutable objects can be used to have the same effect. In one case, you update the same instance of the object (immutable), in the other case the entire object gets replaced with a fresh new instance, so any given instance, after instantiation, never changes value.

Often the immutable object is called an "entity". The main difference between an entity and a value object is that an entity has an ID, and is identifiable.

The configuration systems I know use the latter method, i.e. it is an "anonymous" and disposable object that just carries the desired value, not an entity, with a lifecycle and identity.

I have trouble imagining why one wouldn't use value objects for configuration.

So how do we resolve our differences of opinion?

(BTW, I'm glad I brought it up here, because I think understanding configuration is actually quite important.)

Reply:
I had a second thought and Value Object is more precisely "a datastructure for which identity is its entire content", if 2 value objects has the same content they should be considered not only equals but the same.
So I'm ok with this use of Value Object